Kernel/init.cpp

```diff
@@ -399,15 +399,24 @@ extern "C" int __cxa_atexit(void (*)(void*), void*, void*)
 
 void setup_acpi()
 {
-    bool complete_acpi_disable = KParams::the().has("noacpi");
-    bool dynamic_acpi_disable = KParams::the().has("noacpi_aml");
-    if (complete_acpi_disable) {
+    if (!KParams::the().has("acpi")) {
+        ACPIDynamicParser::initialize_without_rsdp();
+        return;
+    }
+
+    auto acpi = KParams::the().get("acpi");
+    if (acpi.matches("off")) {
```

Comment:
No need to use `String::matches()` here since we're not trying to support globbing.
```cpp
if (acpi == "off") {
```
(And the same comment for every other use of `matches`.)